smack-experimental/src/main/java/org/jivesoftware/smackx/muclight/MultiUserChatLightManager.java

```diff
@@ -0,0 +1,462 @@
+ */
+public final class MultiUserChatLightManager extends Manager {
+
+    private static final Map<XMPPConnection, MultiUserChatLightManager> INSTANCES = new WeakHashMap<XMPPConnection, MultiUserChatLightManager>();
+
+    /**
+     * Get a instance of a MUC Light manager for the given connection.
+     *
+     * @param connection
+     * @return a MUCLight manager.
+     */
+    public static synchronized MultiUserChatLightManager getInstanceFor(XMPPConnection connection) {
+        MultiUserChatLightManager multiUserChatLightManager = INSTANCES.get(connection);
+        if (multiUserChatLightManager == null) {
+            multiUserChatLightManager = new MultiUserChatLightManager(connection);
+            INSTANCES.put(connection, multiUserChatLightManager);
+        }
+        return multiUserChatLightManager;
+    }
+
+    /**
+     * A Map of MUC Light JIDs to instances. We use weak references for the
+     * values in order to allow those instances to get garbage collected.
+     */
+    private final Map<EntityBareJid, WeakReference<MultiUserChatLight>> multiUserChatLights = new HashMap<>();
+
+    private MultiUserChatLightManager(XMPPConnection connection) {
+        super(connection);
+    }
+
+    /**
+     * Obtain the MUC Light.
+     *
+     * @param jid
+     * @return the MUCLight.
+     */
+    public synchronized MultiUserChatLight getMultiUserChatLight(EntityBareJid jid) {
+        WeakReference<MultiUserChatLight> weakRefMultiUserChat = multiUserChatLights.get(jid);
+        if (weakRefMultiUserChat == null) {
+            return createNewMucLightAndAddToMap(jid);
+        }
+        MultiUserChatLight multiUserChatLight = weakRefMultiUserChat.get();
+        if (multiUserChatLight == null) {
+            return createNewMucLightAndAddToMap(jid);
+        }
+        return multiUserChatLight;
+    }
+
+    private MultiUserChatLight createNewMucLightAndAddToMap(EntityBareJid jid) {
+        MultiUserChatLight multiUserChatLight = new MultiUserChatLight(connection(), jid);
+        multiUserChatLights.put(jid, new WeakReference<MultiUserChatLight>(multiUserChatLight));
+        return multiUserChatLight;
+    }
+
+    /**
+     * Returns true if the MUC Light service is enabled.
+     *
+     * @param mucLightService
+     * @return a boolean indicating if the MUC Light service is enabled
+     * @throws XMPPErrorException
+     * @throws NoResponseException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean isServiceEnabled(DomainBareJid mucLightService)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        DiscoverItems discoverItems = ServiceDiscoveryManager.getInstanceFor(connection())
+                .discoverItems(connection().getServiceName());
+        boolean found = false;
+        for (DiscoverItems.Item item : discoverItems.getItems()) {
+            if (item.getEntityID().equals(mucLightService)) {
```

Comment:
That's not very helpful, please remove this method. If you already have a DomainBareJid then you could just perform a disco#info on it and determine if it announces the muc-light feature. A method
```java
public DomainBareJid discoveryService()
```
which returns the DomainBareJid of the local muc-light service or `null` if none was found, would be of much more useful.